### PR TITLE
control-plane flag support for the `kubectl-env` command

### LINKED ISSUE #50: Target Control Plane / Switch from Shoot to Seed
**What would you like to be added**:
It should be possible to switch to the control plane of a shoot or to target it's seed

**Why is this needed**:

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
This PR enhances the `kubectl-env` command to also support if a shoot control-plane or project is targeted.

**Which issue(s) this PR fixes**:
Fixes #50

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: pkg/target/manager.go [hunk 1]
```diff
@@ -417,21 +417,41 @@ func (m *managerImpl) updateTarget(target Target) error {
 }
 
 func (m *managerImpl) ClientConfig(ctx context.Context, t Target) (clientcmd.ClientConfig, error) {
+	if t.ControlPlane() && t.ShootName() != "" {
```
Comment (holgerkoser) [inline pkg/target/manager.go:420]:
```suggestion
	if t.ControlPlane() {
```

File: pkg/target/manager.go [hunk 2]
```diff
@@ -567,3 +603,29 @@ func writeRawConfig(config clientcmd.ClientConfig) ([]byte, error) {
...
+}
+
+func clientConfigWithNamespace(clientConfig clientcmd.ClientConfig, namespace string) (clientcmd.ClientConfig, error) {
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get raw client config: %w", err)
+	}
+
+	contextName := rawConfig.CurrentContext
+	rawConfig.Contexts[contextName].Namespace = namespace
```
Comment (petersutter) [inline pkg/target/manager.go:627]: check if exists (rawConfig.Contexts[contextName])

File: pkg/target/manager.go (hunk 1 above)
Comment (holgerkoser) [reply pkg/target/manager.go]: ✔️

File: pkg/target/manager.go (hunk 2 above)
Comment (holgerkoser) [reply pkg/target/manager.go]: ✔️

Comment (holgerkoser) [reply pkg/target/manager.go]: We now mimify the config and set the contextName on all contexts (e.g. internal and external) belonging to the current cluster.

File: pkg/config/config.go [hunk 3]
```diff
@@ -138,6 +139,46 @@ func (config *Config) ClientConfig(name string) (clientcmd.ClientConfig, error)
...
+	}
+
+	if g.Context != "" {
+		rawConfig.CurrentContext = g.Context
+	}
+
+	// this function returns an error if the currentContext does not exist
+	err = clientcmdapi.MinifyConfig(rawConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to mimify client configuration: %w", err)
```
Comment (petersutter) [inline pkg/config/config.go:176]:
```suggestion
		return nil, fmt.Errorf("failed to minify client configuration: %w", err)
```

**Review by petersutter — APPROVED** [review]
/lgtm apart from the typo

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**